bitmovin/analytics/impressions.ts

```diff
@@ -5,14 +5,30 @@ import {HttpClient} from '../utils/types';
 
 export const impressions = (configuration, httpClient: HttpClient) => {
   const {post} = httpClient;
-  const impressionBaseUrl = urljoin(configuration.apiBaseUrl, 'analytics', 'impressions');
+  const impressionsBaseUrl = urljoin(configuration.apiBaseUrl, 'analytics', 'impressions');
 
-  const fn = (impressionId, licenseKey) => {
-    const url = urljoin(impressionBaseUrl, impressionId);
+  const details = (impressionId: string, licenseKey: string) => {
+    const url = urljoin(impressionsBaseUrl, impressionId);
     return post(configuration, url, {licenseKey});
   };
+  const list = (
```

Comment:
Don't you think we should just make this a `payload` object - we usually only make path parameters function variables and everything else is just an object.
Also we would have to start destructuring the payloads and assigning them to parameter names right?